integration/all/integration_test.go

```diff
@@ -0,0 +1,314 @@
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"testing"
+
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+	timeout                    = 10
+
+	K8sDistroAKSAzureLinux           = "aks-AzureLinux"
+	K8sDistroAKSUbuntu               = "aks-Ubuntu"
+	K8sDistroEKSAmazonLinux          = "eks-AmazonLinux"
+	K8sDistroARO                     = "aro"
+	K8sDistroMinikubeGH              = "minikube-github"
+	securityProfileOperatorNamespace = "security-profiles-operator"
+	topTimeoutInSeconds              = 10
+)
+
+var (
+	supportedK8sDistros = []string{K8sDistroAKSAzureLinux, K8sDistroAKSUbuntu, K8sDistroARO, K8sDistroMinikubeGH}
+	cleaningUp          = uint32(0)
+
+	containerRuntime string
+	isDockerRuntime  bool
+	isCrioRuntime    bool
+)
+
+var (
+	integration    = flag.Bool("integration", false, "run integration tests")
+	dnsTesterImage = flag.String("dnstester-image", "ghcr.io/inspektor-gadget/dnstester:latest", "dnstester container image")
+	testComponent  = flag.String("test-component", "", "run tests for specific component")
+
+	image          = flag.String("image", "", "gadget container image")
+	doNotDeployIG  = flag.Bool("no-deploy-ig", false, "don't deploy Inspektor Gadget")
+	doNotDeploySPO = flag.Bool("no-deploy-spo", false, "don't deploy the Security Profiles Operator (SPO)")
+	k8sDistro      = flag.String("k8s-distro", "", "allows to skip tests that are not supported on a given Kubernetes distribution")
+	k8sArch        = flag.String("k8s-arch", "amd64", "allows to skip tests that are not supported on a given CPU architecture")
+)
+
+func normalizeCommonData(e *eventtypes.CommonData, ns string) {
+	if DefaultTestComponent == IgTestComponent {
+		// Docker and CRI-O use a custom container name composed, among
+		// other things, by the pod UID. We don't know the pod UID in
+		// advance, so we can't match the exact expected container name.
+		prefixContainerName := "k8s_" + "test-pod" + "_" + "test-pod" + "_" + ns + "_"
+		if (containerRuntime == ContainerRuntimeDocker || containerRuntime == ContainerRuntimeCRIO) &&
+			strings.HasPrefix(e.Runtime.ContainerName, prefixContainerName) {
+			e.Runtime.ContainerName = "test-pod"
+		}
+		// Docker can provide different values for ContainerImageName. See `getContainerImageNamefromImage`
+		if isDockerRuntime {
+			e.Runtime.ContainerImageName = ""
+		}
+
+	} else if DefaultTestComponent == InspektorGadgetTestComponent {
+		e.K8s.Node = ""
+		// TODO: Verify container runtime and container name
+		e.Runtime.RuntimeName = ""
+		e.Runtime.ContainerName = ""
+	}
+}
+
+func cleanupFunc(cleanupCommands []*Command) {
+	if !atomic.CompareAndSwapUint32(&cleaningUp, 0, 1) {
+		return
+	}
```

Comment:
This was already part of our code and I think it's working properly. It ensures `cleanupFunc` is executed only once. I think it does the same of `sync.Once` but less readable hahaha. I think we can move to `sync.Once` but in another PR to avoid mixing things. WDYT?